binary_search_tree/tree.py

```diff
@@ -14,46 +14,110 @@ class Tree:
     def __init__(self):
         self.root = None
 
-    # Time Complexity: 
-    # Space Complexity: 
+    #----------------------- ADD ----------------------
+    #from from class video 👇🏼
+    def add_helper(self, current, key, value):
+        if current == None:
+            return TreeNode(key, value)
+        elif current.key >= key:
+            current.left = self.add_helper(current.left, key, value)
+        else: 
+            current.right = self.add_helper(current.right, key, value)
+        return current 
```

Comment:
This implementation of add_helper works, but definitely does a bit of extra work in setting nodes. Essentially, every node on the path to where the new node ends up will have either its left or right node "updated", though most of these will be updated to the exact same node. The only one that gets a meaningful update is whenever the bottom of the tree is reached and gets its left/right node updated to the new node. These updates happen in lines 23 and 25.

This probably will not be too bad of a performance hit, but there is a way that you can do this and just update only the node you need.

But this overall fine enough!